### PR TITLE
fix: stargate tests against online db

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
potential fix for testing against online db

### CODE REVIEW

File: test/step-definitions/requests.ts [hunk 1]
```diff
@@ -19,14 +19,16 @@ export class requests {
...
   public async before(): Promise<void> {
-    const options: cliOptions = {
-      namespaceUrl: 'http://localhost:8082/v2/schemas/namespaces',
-      authUrl: process.env.STARGATE_AUTH_URL,
-      keyspace: 'eddiehub',
-      password: 'cassandra',
-      username: 'cassandra',
-    };
-    await setup(options);
+    if (process.env.STARGATE_BASEURL) {
```
Comment (Cahllagerfeld) [inline test/step-definitions/requests.ts:22]: I think this resolves the problem

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]: `options` will be undefined, I think we will need to pass in other options for Astra in the cloud

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]:
```ts
✗ npm run test:e2e 

> api@0.5.2 test:e2e /Users/ej/Downloads/repos/api
> cucumber-js -p default

(node:6148) UnhandledPromiseRejectionWarning: Error: please define keyspace name
    at Object.<anonymous> (/Users/ej/Downloads/repos/api/node_modules/@cahllagerfeld/setup-stargate/src/setup.ts:9:11)
    at step (/Users/ej/Downloads/repos/api/node_modules/@cahllagerfeld/setup-stargate/dist/setup.js:33:23)
    at Object.next (/Users/ej/Downloads/repos/api/node_modules/@cahllagerfeld/setup-stargate/dist/setup.js:14:53)
    at /Users/ej/Downloads/repos/api/node_modules/@cahllagerfeld/setup-stargate/dist/setup.js:8:71
```

Comment (Cahllagerfeld) [reply test/step-definitions/requests.ts:22]: When doing it with the online DB we dont need any configuration.
```ts
 authUrl: process.env.STARGATE_AUTH_URL
          ? process.env.STARGATE_AUTH_URL
          : '',
```
Would something like this help`?

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]: I will take a closer look Sunday morning

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]: Actually the error is ...

```ts
               _                                   _                                    _          
  ___    ___  | |_   _   _   _ __            ___  | |_    __ _   _ __    __ _    __ _  | |_    ___ 
 / __|  / _ \ | __| | | | | | '_ \   _____  / __| | __|  / _` | | '__|  / _` |  / _` | | __|  / _ \
 \__ \ |  __/ | |_  | |_| | | |_) | |_____| \__ \ | |_  | (_| | | |    | (_| | | (_| | | |_  |  __/
 |___/  \___|  \__|  \__,_| | .__/          |___/  \__|  \__,_| |_|     \__, |  \__,_|  \__|  \___|
                            |_|                                         |___/                      
couldn't create stargate namespace
    ✖ failed
      Error: Request Failed: undefined
      Stack Trace: connect ECONNREFUSED 127.0.0.1:8081
          at axiosRequest (/Users/ej/Downloads/repos/api/node_modules/@astrajs/rest/src/rest.js:126:11)
          at processTicksAndRejections (internal/process/task_queues.js:93:5)
          at Object.createClient (/Users/ej/Downloads/repos/api/node_modules/@astrajs/rest/src/rest.js:63:24)
          at Object.createClient (/Users/ej/Downloads/repos/api/node_modules/@astrajs/collections/src/collections.js:146:22)
```

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]: Fresh pair of eyes in the morning helped. I got the tests passing by needing the namespace/keyspace on the cloud.

Comment (eddiejaoude) [reply test/step-definitions/requests.ts:22]: But I also had to comment out the `import` not sure why it is being triggered

```
import { setup, cliOptions } from '@cahllagerfeld/setup-stargate';
```

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
   },
   "dependencies": {
     "@cahllagerfeld/nestjs-astra": "0.0.2",
-    "@cahllagerfeld/setup-stargate": "0.0.10",
+    "@cahllagerfeld/setup-stargate": "^1.0.0",
```
Comment (Cahllagerfeld) [inline package.json:28]: The failure was in main.ts of my package. 
I updated this

File: test/step-definitions/requests.ts [hunk 3]
```diff
@@ -19,14 +19,9 @@ export class requests {
...
```
Comment (Cahllagerfeld) [inline test/step-definitions/requests.ts:22]: I updated the script to not import the functions, instead running our prestart script

File: package.json [hunk 4]
```diff
@@ -6,8 +6,8 @@
   "license": "MIT",
   "scripts": {
     "prebuild": "rimraf dist",
-    "prestart": "setup-stargate -k eddiehub",
     "build": "nest build",
+    "prestart": "setup-stargate default eddiehub",
```
Comment (Cahllagerfeld) [inline package.json:10]: adjusted the prestartscript

**Review by Cahllagerfeld — COMMENTED** [review]
This worked for me locally and with stargate

**Comment by eddiejaoude** [comment]
Looks good 👍 , for some reason I get `500` errors now 🤔 

```ts
  Scenario: get list of events
    Given make a GET request to "/calendar"
[Nest] 20417   - 06/06/2021, 17:27:39   [ExceptionHandler] Request Failed: [object Object]
Stack Trace: Request failed with status code 500
Error: Request Failed: [object Object]
Stack Trace: Request failed with status code 500
    at axiosRequest (/Users/ej/Downloads/repos/api/node_modules/@astrajs/rest/src/rest.js:126:11)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at AstraClient._request (/Users/ej/Downloads/repos/api/node_modules/@astrajs/rest/src/rest.js:199:22)
    at AstraClient.get (/Users/ej/Downloads/repos/api/node_modules/@astrajs/rest/src/rest.js:226:12)
    at AstraCollection._get (/Users/ej/Downloads/repos/api/node_modules/@astrajs/collections/src/collections.js:61:22)
    at AstraCollection.find (/Users/ej/Downloads/repos/api/node_modules/@astrajs/collections/src/collections.js:84:12)
    Then the response status code should be 200
    ✖ failed
      AssertionError
          + expected - actual
      
          -500
          +200
      
          at responses.statusResponse (/Users/ej/Downloads/repos/api/test/step-definitions/responses.ts:28:45)
          at World.bindingFunc (/Users/ej/Downloads/repos/api/node_modules/cucumber-tsflow/src/binding-decorator.ts:172:22)

```

Did you have to do anything else, like envars etc?

**Comment by Cahllagerfeld** [comment]
Is this for testing locally or online?

I just came to the thought that online tests wont work: 
 When testing the `Get-All`-Endpoint for any the tests will always fail, except the db is compleltly unsed

**Comment by eddiejaoude** [comment]
This is for testing locally against the cloud. Not something that would be used often or we would recommend, but it would lower the barrier to entry for people who do not have docker installed. 

Why would the seeding not work? I thought the data is seeded via the API, so it would work the same?

**Comment by eddiejaoude** [comment]
I think the `500` is because there is no data in the DB and we do not handle the `404` exception. We will need something like this...

```ts
findAll(): Observable<findResult<DiscordProfile>> {
      return this.astraService.find<DiscordProfile>()
        .pipe(
          catchError((e) => from([]))
        );
  }
```

**Comment by eddiejaoude** [comment]
I think this PR is ready to merge?

**Comment by Cahllagerfeld** [comment]
Let me check it again after work before we merge

**Comment by eddiejaoude** [comment]
Sure 👍

**Comment by eddiejaoude** [comment]
I created a PR with the changes that contains your patch from this PR https://github.com/EddieHubCommunity/api/pull/148

**Comment by eddiejaoude** [comment]
I will close these changes are now in https://github.com/EddieHubCommunity/api/pull/148

**Pull request closed without merging**